### DISCUSSION TITLE
Tailwind theme config from my React components library not picked up by main app's Tailwind

### DISCUSSION THREAD

**Discussion by rohvald** [tailwindlabs/tailwindcss#11120] [Help] [answered]
I have a React components library and a NextJS app. They both use TailwindCSS. I'm using Nx monorepo, in case that has a thing to play here, but I highly doubt it.

The app uses components from the library and renders the components well, except for the theme configurations from tailwind.config.js file of the library.

What am I missing?

Lib's tailwind.config.js:
```
module.exports = {
  content: [
    'libs/moonlight-neon-ui/src/**/*.{html,js,jsx,tsx,ts}'
  ],
  theme: {
    extend: {
      fontFamily: {
        jost: ['Jost', 'Poppins', 'Verdana', 'sans-serif'],
      },
      colors: {
        'background': '#141414',

        'hlt': '#5f5f5f9e',
        'hlb': '#121212',
        'bgt': '#202227',
        'bgb': '#3b3d43',
        'bght': '#2a2c33',
        'bghb': '#43464d',
        
        'cbg': '#1c1e23',
        'cbgb': '#24262a',

        'glow': '#fbb615bf',

        'p1': '#f26e25',
        'p1h': '#f26e259C',
        'p2': '#fbb615',
        'p2h': '#fbb6159C',
        'pb': '#1a1a1a',
        'pw': '#f3f3f3',

        's1': '#ef3e23',
        's2': '#f37b21',
        'sb': '#231f20',
        'sw': '#fff',

        // 979797 > 500
        // 262626 > 900
      }
    },
  },
  plugins: [],
};
```

App's tailwind.config.js:

```
/** @type {import('tailwindcss').Config} */
module.exports = {
  // presets: [
  //   require('libs/moonlight-neon-ui/tailwind.config.js')
  // ],
  content: [
    'apps/ac-dcp/**/*.{html,js,jsx,tsx,ts}',
    'libs/moonlight-neon-ui/**/*.{html,js,jsx,tsx,ts}'
  ],
  theme: {
    extend: {}
  },
  plugins: [],
}
```

**Comment by wongjn** [comment, marked as answer]
You seem to have commented out the `presets` configuration in the app's `tailwind.config.js` so it has no effect. Consider uncommenting it like so:
```diff
/** @type {import('tailwindcss').Config} */
module.exports = {
-  // presets: [
-  //   require('libs/moonlight-neon-ui/tailwind.config.js')
-  // ],
+  presets: [
+     require('libs/moonlight-neon-ui/tailwind.config.js')
+  ],
   content: [
     'apps/ac-dcp/**/*.{html,js,jsx,tsx,ts}',
```

**Reply by rohvald** [reply]
Thanks for replying on stackoverflow too, one of the relative paths worked afterall and yes, i uncommented this and tried to make it work again. So happy this is over, lol. Thanks for the help!

**Comment by saidinusah** [comment]
hello @rohvald , my component library styles are not showing when i use them, please is there anything i may be doing wrong?

https://github.com/tailwindlabs/tailwindcss/discussions/11172

/** @type {import('tailwindcss').Config} */
export default {
  content: ["./index.html", "./src/**/*.{js,ts,jsx,tsx}"],
  theme: {
    extend: {
      borderRadius: {
        sm: "0.25rem",
        md: "0.5rem",
      },
      colors: {
        primary: {
          "p-50": "#EBF5FF",
          "p-100": "#CDE7FE",
          "p-200": "#91CBFD",
          "p-300": "#4BA9FC",
          "p-500": "#0476D9",
          "p-600": "#035296",
        },
        gray: {
          "g-25": "#F6F8F8",
          "g-50": "#F1F3F4",
          "g-100": "#DCE1E4",
          "g-200": "#BDC5CC",
          "g-400": "#97A4AF",
          "g-400": "#6F8190",
          "g-500": "#505D68",
          "g-600": "#384149",
        },
        error: {
          "e-50": "#FEF3F2",
          "e-100": "#FEE4E2",
          "e-200": "#FECDCA",
          "e-300": "#FDA29B",
          "e-500": "#F04438",
          "e-600": "#912018",
        },
        success: {
          "s-100": "#DCF9E7",
          "s-200": "#BEF4D2",
          "s-300": "#92ECB3",
          "s-400": "#62E492",
          "s-500": "#22C55E",
          "s-600": "#188B42",
        },
        warning: {
          "w-50": "#FFFAEB",
          "w-100": "#FEF0C7",
          "w-200": "#FEDF89",
          "w-300": "#FDB022",
          "w-500": "#F79009",
          "w-600": "#93370D",
        },
        black: "#000",
        white: "#fff",
      },
    },
  },
  prefix: "church-pro-",
};

**Comment by rohvald** [comment]
1. If this is a copy of tailwind.config.js from your lib, make sure that content includes all the files you need styled.
2. Add the lib config and components to the tailwind.config.js of the app. For some reason i had to use a relative path instead of an absolute for the presets: [ require('...') ]. My example:
```
/** @type {import('tailwindcss').Config} */
module.exports = {
  presets: [
    require('../../libs/moonlight-neon-ui/tailwind.config.js'),
  ],
  content: [
    'apps/ac-dcp/**/*.{html,js,jsx,tsx,ts}',
    'libs/moonlight-neon-ui/src/**/*.{html,js,jsx,tsx,ts}'
  ],
  theme: {
    extend: {}
  },
  plugins: [],
}
```
